common/scala/src/main/scala/org/apache/openwhisk/core/entity/Attachments.scala

```diff
@@ -66,13 +66,11 @@ object Attachments {
     implicit val serdes = {
       implicit val contentTypeSerdes = new RootJsonFormat[ContentType] {
         override def write(c: ContentType) = JsString(c.value)
-        override def read(js: JsValue) =
-          Try {
-            val JsString(c) = js
-            ContentType.parse(c).right.get
-          } getOrElse {
-            throw new DeserializationException("Could not deserialize content-type")
-          }
+        override def read(js: JsValue) = {
+          Try(js.convertTo[String])
+            .flatMap(ContentType.parse(_).fold(_ => Failure(new Exception("failed to parse content-type")), Success(_)))
```

Comment:
could this be further simplified to map to success only? we're not using the exception from left.

Reply:
Left doesn't have an exception but a custom error class. To fold everything into a Try, the left needs to be filled with an Exception or we'd need to throw in two places alternatively.

Reply:
thanks - nm then.